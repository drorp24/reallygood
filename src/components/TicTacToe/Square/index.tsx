// @ts-nocheck

import React, { memo } from "react";

import styles from "./styles.module.scss";

// I sometimes assume there could be some more ('...rest') arguments that I need to pass onwards to the child
// The reason I did it here is that in a real app I would want to highlight the 'winning squares'.
// To do that I would need to control the styling of the Square component from its parent
// and '...rest' could help in that.
export default memo(function Square({ value, onClick, ...rest }): JSX.Element {
  // TODO: fix...
  const color = value === "X" ? "#50a1d2" : "#333";
  return (
    <div className={styles.Square} style={{ color }} {...{ onClick, ...rest }}>
      {value}
    </div>
  );
});
