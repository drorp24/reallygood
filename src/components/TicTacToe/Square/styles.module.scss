.Square {
  width: 100%;
  height: 100%;
  margin: 4px;
  display: flex;
  justify-content: center;
  align-items: center;
  color: #50a1d2;
  font-size: 14vw;
}
