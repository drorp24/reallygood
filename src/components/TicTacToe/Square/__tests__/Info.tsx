import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";

import Square from "../index";

describe("<Square />", () => {
  it("renders", () => {
    const tree = render(
      <MemoryRouter>
        <Square />
      </MemoryRouter>
    ).container.firstChild;

    expect(tree).toMatchSnapshot();
  });
});
