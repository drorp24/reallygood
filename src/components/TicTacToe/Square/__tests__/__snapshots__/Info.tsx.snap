// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Square /> renders 1`] = `
<div
  class="Square"
/>
`;
